packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -0,0 +1,286 @@
+  readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /** The default EC2 instance type to use for databases in this cluster. */
+  readonly defaultInstanceType: ec2.InstanceType;
+
+  /**
+   * Returns a new instance of this engine with a particular version specified.
+   * Calling {@link engineVersion} on the returned instance will always return the provided value.
+   *
+   * @param engineVersion the exact engine version, for example "5.7.mysql_aurora.2.03.4"
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-engineversion
+   */
+  withVersion(engineVersion: string): IClusterEngine;
+
+  /**
+   * Method called when the engine is used to create a new cluster.
+   */
+  bindToCluster(scope: core.Construct, options: ClusterEngineBindOptions): ClusterEngineBindConfig;
+}
+
+abstract class AbstractClusterEngine implements IClusterEngine {
+  public readonly engineType: string;
+  public readonly engineVersion?: string;
+  public readonly parameterGroupFamily: string;
+  public readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+  public readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+  // this works for all 3 cluster engines that are currently supported
+  public readonly defaultInstanceType = ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.MEDIUM);
+
+  private readonly needsS3RolesInParameters: boolean;
+  private readonly parameterGroupFamilies?: ParameterGroupFamilyMapping[];
+  private readonly defaultPort?: number;
+
+  constructor(
+    engineName: string,
+    needsS3RolesInParameters: boolean,
+    singleUserRotationApplication: secretsmanager.SecretRotationApplication,
+    multiUserRotationApplication: secretsmanager.SecretRotationApplication,
+    parameterGroupFamilies?: ParameterGroupFamilyMapping[],
+    defaultPort?: number,
+    engineVersion?: string) {
+
+    this.engineType = engineName;
+    this.needsS3RolesInParameters = needsS3RolesInParameters;
+    this.singleUserRotationApplication = singleUserRotationApplication;
+    this.multiUserRotationApplication = multiUserRotationApplication;
+    this.parameterGroupFamilies = parameterGroupFamilies;
+    this.defaultPort = defaultPort;
+    this.engineVersion = engineVersion;
+    this.parameterGroupFamily = this.establishParameterGroupFamily();
+  }
+
+  public withVersion(engineVersion: string): IClusterEngine {
+    const self = this;
+    // put it in a local variable, as it's protected
+    const defaultParameterGroup = this.defaultParameterGroup;
+
+    class CopyWithVersion extends AbstractClusterEngine {
+      protected defaultParameterGroup(scope: core.Construct) {
+        // we need to use call, so that 'this' inside defaultParameterGroup() is resolved correctly
+        return defaultParameterGroup.call(self, scope);
```

Comment:
Holy cow! There should have been a way to do this more easily.

Reply:
Meh, doesn't matter that much, it's all encapsulated inside the class anyway.

And we'll probably get rid of the `withVersion` method anyway in favor of having the versions statically on `DatabaseClusterEngine` directly.